app/models/bulkrax/oai_entry.rb

```diff
@@ -62,16 +61,13 @@ def collections_created?
     def find_or_create_collection_ids
       return self.collection_ids if collections_created?
 
-      if sets.blank?
-        c = Collection.where(Bulkrax.system_identifier_field => parser.collection_name).first
+      if sets.blank? || parser.collection_name != 'all'
+        c = Collection.where(Bulkrax.system_identifier_field => importer.unique_collection_identifier(parser.collection_name)).first
+        self.collection_ids << c.id unless c.blank? || self.collection_ids.include?(c.id)
```

Comment:
Let’s break this up in to more lines. I _think_ you are saying
```
if c.present? && !self.collection_ids.include?(c.id)
  self.collection_ids << c.id
end
```

But I got it wrong the first time and that makes me think I’ll get it wrong in the future ;-)